src/app/my-payment-methods/my-payment-methods.component.ts

```diff
@@ -0,0 +1,157 @@
+import {UpdateCardModalComponent} from "../update-card-modal/update-card-modal.component";
+import {HttpErrorResponse} from "@angular/common/http";
+import {isPlatformBrowser} from "@angular/common";
+import {Person} from "../person.model";
+import {ExactCurrencyPipe} from "../exact-currency.pipe";
+import {MatProgressSpinner} from "@angular/material/progress-spinner";
+import {DonationService} from "../donation.service";
+import {IdentityService} from "../identity.service";
+import {MatDialog} from "@angular/material/dialog";
+
+@Component({
+  selector: 'app-my-payment-methods',
+  standalone: true,
+  imports: [
+    ComponentsModule,
+    ExactCurrencyPipe,
+    MatProgressSpinner
+  ],
+  templateUrl: './my-payment-methods.component.html',
+  styleUrl: './my-payment-methods.component.scss'
+})
+export class MyPaymentMethodsComponent implements OnInit, OnDestroy{
+  protected paymentMethods: PaymentMethod[] | undefined;
+  protected person: Person;
+
+  private savedCardsTimer: undefined | ReturnType<typeof setTimeout>; // https://stackoverflow.com/a/56239226
+
+  registerErrorDescription: string | undefined;
+  registerSuccessMessage: string | undefined;
+  constructor(
+    private donationService: DonationService,
+    private identityService: IdentityService,
+    private route: ActivatedRoute,
+    public dialog: MatDialog,
+    @Inject(PLATFORM_ID) private platformId: Object,
+  ) {}
+
+  ngOnInit(): void {
+    this.paymentMethods = this.route.snapshot.data.paymentMethods;
+    this.person = this.route.snapshot.data.person;
+    if (isPlatformBrowser(this.platformId)) {
+       this.savedCardsTimer = setTimeout(this.checkForPaymentCardsIfNotLoaded, 5_000);
+    }
+  }
+
+  protected get hasSavedPaymentMethods()
+  {
+    return this.paymentMethods !== undefined && this.paymentMethods.length > 0;
+  }
+
+  /**
+   * To work around a bug where it seems sometimes new payment cards are not showing on this page, if there are none
+   * loaded at this point we check again with the server.
+   */
+  private checkForPaymentCardsIfNotLoaded = async () => {
+    if (this.paymentMethods && this.paymentMethods.length > 0) {
+      return;
+    }
+
+    this.paymentMethods = await this.donationService.getPaymentMethods({cacheBust: true})
+  };
+
+  async loadPaymentMethods() {
+    // not so keen on the component using the donation service and the identity service together like this
+    // would rather call one service and have it do everything for us. Not sure what service would be best to put
+    // this code in.
+    this.paymentMethods = await this.donationService.getPaymentMethods({cacheBust: true})
+  }
+
+  deleteMethod(method: PaymentMethod) {
+    this.paymentMethods = undefined;
+
+    this.donationService.deleteStripePaymentMethod(this.person, method, this.jwtAsString()).subscribe(
+      this.loadPaymentMethods.bind(this),
+      error => {
+        this.loadPaymentMethods.bind(this)()
+        alert(error.error.error)
```

Comment:
Not new code, but would ideally probably be better to use a toast or print error somewhere. Similarly, I think `registerErrorDescription` below may not be rendered anywhere?